src/node/mod.rs

```diff
@@ -359,12 +359,17 @@ pub trait AudioScheduledSourceNode: AudioNode {
     /// determined by stop() is reached. For an [`AudioBufferSourceNode`], the event is also
     /// dispatched because the duration has been reached or if the entire buffer has been played.
     ///
-    /// Calling this function multiple times will accumulate all event handlers. It is currently
-    /// not possible to remove an event handler.
-    fn onended<F: FnOnce() + Send + 'static>(&self, callback: F) {
-        self.context().register_event_handler(
+    /// Only a single event handler is active at any time. Calling this method multiple times will
+    /// override the previous event handler.
+    fn onended<F: FnOnce() + Send + 'static>(&self, value: Option<F>) {
+        let callback = value.map(|f| {
```

Comment:
Notice I'm using a silly trick here and [here](https://github.com/orottier/web-audio-api-rs/pull/245/files#diff-d20ac8af5de1b3b89086e74e5f5881a19f0d73a14819c79546277cfbb9f1ecc9R176)
The user can supply a `FnOnce()` here which takes no arguments. Which is good because the `onended` event carries no payload.
For the AudioRenderCapacity, the user supplies `FnMut(AudioRenderCapacityEvent)` because there is a payload here.
Since all callbacks live in the `event_handlers` map of the `EventLoop`, I'm converting them both to take a payload and erase these differences here